### PR TITLE
Uid2 decouple cstg

### PR DESCRIPTION
**Pull request by jingyi-gao-ttd**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

File: modules/uid2Cstg.js [hunk 1]
```diff
@@ -0,0 +1,490 @@
+/* eslint-disable no-console */
```
Comment (mcollins-ttd) [inline modules/uid2Cstg.js]: Don't think we need this, can't see any uses of console in this file

File: modules/uid2IdSystem.js [hunk 2]
```diff
@@ -117,8 +118,13 @@ export const uid2IdSubmodule = {
       }
     },
   },
+
 };
 
+export function attachTokenGenerator(tokenGeneratorModule) {
```
Comment (mcollins-ttd) [inline modules/uid2IdSystem.js]: Does a publisher who wishes to use CSTG need to call this, or is it just for tests?

Comment (jingyi-gao-ttd) [reply modules/uid2IdSystem.js]: No, the publisher don't need to call this, this is for async submodule loading. no matter which module loads first, prebid core will call this function and pass the uid2Cstg in here.

https://github.com/prebid/Prebid.js/blob/master/src/hook.js#L36C8-L49

File: package.json [hunk 3]
```diff
@@ -131,6 +131,7 @@
     "dset": "3.1.2",
     "express": "^4.15.4",
     "fun-hooks": "^0.9.9",
+    "gulp-cli": "^2.3.0",
```
Comment (mcollins-ttd) [inline package.json]: Should this be a dev dependency?

File: gulpfile.js [hunk 4]
```diff
@@ -450,7 +451,13 @@ gulp.task('build-bundle-verbose', gulp.series(makeWebpackPkg({
 // public tasks (dependencies are needed for each task since they can be ran on their own)
 gulp.task('test-only', test);
 gulp.task('test-all-features-disabled', testTaskMaker({disableFeatures: require('./features.json'), oneBrowser: 'chrome', watch: false}));
+// gulp.task('test-watch', testTaskMaker({watch: true, file: ['test/spec/modules/uid2idsystem_spec.js']}));
```
Comment (mcollins-ttd) [inline gulpfile.js]: Do we want to keep the commented-out lines?

File: integrationExamples/gpt/userId_example.html
Comment (mcollins-ttd) [inline integrationExamples/gpt/userId_example.html:1]: It looks like this example isn't specific to UID2, can we make such a big change?

File: karma.conf.maker.js
Comment (mcollins-ttd) [inline karma.conf.maker.js:1]: Do we want to keep the commented-out lines?

File: integrationExamples/gpt/userId_example.html
Comment (jingyi-gao-ttd) [reply integrationExamples/gpt/userId_example.html:1]: No, this PR is against my original PR and not pushed to Prebid repo, those changes are easy for everyone to pull this branch and run tests to see the coverage and tests. I will revert this back when I merge back to [uid2-cstg-integration](https://github.com/jingyi-gao-ttd/Prebid.js/tree/uid2-cstg-integration)

File: karma.conf.maker.js
Comment (jingyi-gao-ttd) [reply karma.conf.maker.js:1]: Same as above, with this commented-out when you spin up `gulp serve-fast` you will only see uid2 in the uid2_example page. Those will be removed once we merge back to [uid2-cstg-integration](https://github.com/jingyi-gao-ttd/Prebid.js/tree/uid2-cstg-integration)

File: package.json (hunk 3 above)
Comment (jingyi-gao-ttd) [reply package.json]: good catch, I will remove it

File: modules/uid2IdSystem.js [hunk 5]
```diff
@@ -83,21 +83,21 @@ export const uid2IdSubmodule = {
...
     }
 
-    let mappedCstgConfig = {
-      serverPublicKey: config?.params?.serverPublicKey,
-      subscriptionId: config?.params?.subscriptionId,
-      ...extractIdentityFromParams(config?.params ?? {})
-    }
-
-    const mappedConfig = {
+    let mappedConfig = {
```
Comment (mcollins-ttd) [inline modules/uid2IdSystem.js]: This can still be const

File: modules/uid2IdSystem_shared.js [hunk 6]
```diff
@@ -624,26 +706,26 @@ export function Uid2GetId(config, prebidStorageManager, _logInfo, _logWarn) {
     }
   }
 
-  if (cstgIdentity) {
-    if (storedTokens) {
-      const identity = Object.values(cstgIdentity)[0]
-      if (!storedTokenFromSameIdentity(storedTokens, identity)) {
-        _logInfo('CSTG supplied new identity - ignoring stored value.', storedTokens.originalIdentity, cstgIdentity);
-        // Stored token wasn't originally sourced from the provided identity - ignore the stored value. A new user has logged in?
+  if (FEATURES.UID2_CSTG && isCstgEnabled) {
```
Comment (mcollins-ttd) [inline modules/uid2IdSystem_shared.js:709]: Do we need to check `FEATURES.UID2_CSTG` here?

File: modules/uid2IdSystem_shared.js [hunk 7]
```diff
@@ -670,13 +752,9 @@ export function Uid2GetId(config, prebidStorageManager, _logInfo, _logWarn) {
     originalToken: suppliedToken ?? storedTokens?.originalToken,
     latestToken: newestAvailableToken,
   };
-  if (cstgIdentity) tokens.originalIdentity = storedTokens?.originalIdentity
+  if (FEATURES.UID2_CSTG && isCstgEnabled) {
```
Comment (mcollins-ttd) [inline modules/uid2IdSystem_shared.js:755]: Do we need to check `FEATURES.UID2_CSTG` here?

**Review by mcollins-ttd — APPROVED** [review]

File: modules/uid2IdSystem_shared.js (hunk 7 above)
Comment (jingyi-gao-ttd) [reply modules/uid2IdSystem_shared.js:755]: we could remove it, with `FEATURES.UID2_CSTG` line 709-726 will be removed from build, just reduce some bytes

**Pull request merged**